services/osparc-gateway-server/src/osparc_gateway_server/backend/utils.py

```diff
@@ -344,7 +345,12 @@ async def get_cluster_information(docker_client: Docker) -> ClusterInformation:
 async def get_next_empty_node_hostname(
     docker_client: Docker, cluster: Cluster
 ) -> Hostname:
-    cluster_nodes = await docker_client.nodes.list()
+    if not hasattr(get_next_empty_node_hostname, "counter"):
```

Comment:
Minor: You could store that information also in the `cluster`, no? Then it would even be persistent. But I guess if the gateway is down we anyway lose all the jobs